lab_2/main.py

```diff
@@ -6,14 +6,29 @@
 from lab_1.main import tokenize, remove_stop_words
 
 
-# 4
 def get_freq_dict(tokens: list) -> dict or None:
     """
     Calculates frequencies of given tokens
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
-    pass
+    if not isinstance(tokens, list):
+        return None
+    for token in tokens:
+        if not isinstance(token, str):
+            return None
+
+    freqs = {}
+    full_length = len(tokens)
+    for word in tokens:
+        if word not in freqs.keys():
+            freqs[word] = 1
+        else:
+            freqs[word] += 1
+
+    freqs.update((x, round(y / full_length, ndigits=5)) for x, y in freqs.items())
```

Comment:
what happens here?

Reply:
I am changing the values of the keys from raw frequencies (like 5 if let's say there are 10 words) to freq / all elements resulting in asked value (like 5/10 in my example)